src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -1554,12 +1554,12 @@ protected void loadMoreGuaranteed() throws IOException {
     // Can't declare as deprecated, for now, but shouldn't be needed
     protected void _finishString() throws IOException { }
 
-    protected final void createChildArrayContext(final int lineNr, final int colNr) {
+    protected final void createChildArrayContext(final int lineNr, final int colNr) throws JsonParseException {
```

Comment:
With `JsonParser`, Jackson 2.x, I have moved mostly to declaring `IOException` so let's do that with this method and everything else -- there were often cases where some methods need to throw `IOException` (ones that may need to load more content) and then there were mismatches. External API of `JsonParser` exposes it anyway so there is no benefit from internal methods from trying to be more selective.

Reply:
changed